### ISSUE TITLE
randperm is only implemented for cpu when i train pytorch model.Who can solve it ,thx much.

### ISSUE THREAD

**Issue by jaychou790118** [pytorch/pytorch#28415]
## To Reproduce

1.
1.
1.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by fmassa** [comment]
`torch.randperm` is implemented in the GPU as well
```python
In [31]: torch.randperm(5, device='cuda')
Out[31]: tensor([1, 4, 2, 3, 0], device='cuda:0')
```
Make sure you have a recent enough version of PyTorch, the available is present in PyTorch 1.2 and 1.3 at least

**Comment by jaychou790118** [comment]
Thx bro.




| |
13032926926
|
|
邮箱13032926926@163.com
|

Signature is customized by Netease Mail Master

On 10/23/2019 19:37, Francisco Massa wrote:

Closed #28415.
